Avoid incorrect cast in sql_js in wasm builds

ebraminio opened the pull request:
In order to avoid:

    Type 'double' is not a subtype of type 'int' in type cast

Raised when drift is built with `flutter run -d chrome --wasm` it's needed to first cast the value to double then turn it into int and it does makes sense as int isn't native type in JS while double is the default numeric type in JS.

Fixing this make these errors in an external project (stream-chat) go:

<img width="518" height="140" alt="image" src="https://github.com/user-attachments/assets/d6325b13-d63b-4078-8a2d-68e32428b5df" />

ebraminio committed:
Avoid incorrect cast in sql_js in wasm builds

In order to avoid:

    Type 'double' is not a subtype of type 'int' in type cast

Raised when drift is built with `--wasm` it's needed to first cast
the value to double then turn it into int and it does makes
sense as int isn't native type in JS while double is the default
numeric type in JS.

simolus3 commented on drift/lib/src/web/sql_js.dart:
```diff
@@ -114,7 +114,8 @@ class SqlJsDatabase {
   /// Returns the `user_version` pragma from sqlite.
   int get userVersion {
-    return _selectSingleRowAndColumn('PRAGMA user_version;') as int;
+    return (_selectSingleRowAndColumn('PRAGMA user_version;') as double)
```
```suggestion
    return (_selectSingleRowAndColumn('PRAGMA user_version;') as num)
```

simolus3 commented on drift/lib/src/web/sql_js.dart:
```diff
@@ -152,7 +153,8 @@ class SqlJsDatabase {
   int lastInsertId() {
     // load insert id. Will return [{columns: [...], values: [[id]]}]
-    return _selectSingleRowAndColumn('SELECT last_insert_rowid();') as int;
+    return (_selectSingleRowAndColumn('SELECT last_insert_rowid();') as double)
```
```suggestion
    return (_selectSingleRowAndColumn('SELECT last_insert_rowid();') as num)
```

simolus3 reviewed (commented):
Let's use `num` here, apart from that this looks good to me.

ebraminio committed:
Use a cast to num instead of cast to double

Per @simolus3 review, it's better to cast to a more generic type
to reduce confusion.

ebraminio commented:
> Let's use num here, apart from that this looks good to me.

Makes sense. Applied in a separate commit.

ebraminio committed:
Resolve dartfmt complain

ebraminio commented:
Resolved dartfmt complain which was existed before this change AFAICS
